core/src/ipld.rs

```diff
@@ -53,21 +54,85 @@ impl<'a> From<&'a str> for IpldIndex<'a> {
     }
 }
 
+/// Ipld type.
+#[derive(Debug)]
+pub enum IpldType {
```

Comment:
This sounds very similar what is described as `Node` in the [IPLD Library Design docs](https://github.com/ipld/specs/blob/4eb7b233f38864262c24cc91203098f483f21432/design/libraries/nodes-and-kinds.md#node). So perhaps it would make sense to also introduce the concept of a `Node`.

It would also be good to have some more docs about what differentiates this type from `Ipld`.

Reply:
This type is only used for constructing `TypeError`.